validator-frontend/screens/ScanScreen.js

```diff
@@ -33,7 +52,109 @@ export default class ScanScreen extends React.Component {
   }
 
   handleBarCodeScanned = ({ type, data }) => {
-    uncompressedData = pako.inflate(data, { 'to': 'string' })
-    alert(`Bar code with type ${type} and data ${uncompressedData} has been scanned!`);
+    this._storeData(data)
```

Comment:
We store the data before validation is done, that seems kind of fishy.

Reply:
This will be moved to the validating screen in case the qr has been verified